Core/Tasks/InternalCommsTask.c

```diff
@@ -40,30 +43,51 @@ PRIVATE void InternalCommsTask(void *argument)
 	uint32_t cycleTick = osKernelGetTickCount();
 	DebugPrint("icomms");
 	//ICommsInit();
-	uint8_t testTxCounter = 0;
+//	uint8_t testTxCounter = 0;
+
+	uint8_t throttleTxCounter = 0;
+
 	IComms_Init();
 	for(;;)
 	{
 		cycleTick += TIMER_INTERNAL_COMMS_TASK;
 		osDelayUntil(cycleTick);
 		//DebugPrint("icomms loop");
-		iCommsMessage_t txMsg;
-		txMsg.standardMessageID = 0x08;
-		txMsg.dataLength = 4;
-		txMsg.data[0] = 0;
-		txMsg.data[1] = 0;
-		txMsg.data[2] = 0;
-		txMsg.data[3] = 4;
+//		iCommsMessage_t txMsg;
+//		txMsg.standardMessageID = 0x08;
+//		txMsg.dataLength = 4;
+//		txMsg.data[0] = 0;
+//		txMsg.data[1] = 0;
+//		txMsg.data[2] = 0;
+//		txMsg.data[3] = 4;
 		/*txMsg.data[4] = 5;
 		txMsg.data[5] = 6;
 		txMsg.data[6] = 7;
 		txMsg.data[7] = 8;*/
-		testTxCounter++;
-		if(testTxCounter == 25)
-		{
-			DebugPrint("#ICT: Sending!");
-			IComms_Transmit(&txMsg);
-			testTxCounter =0;
+//		testTxCounter++;
+//		if(testTxCounter == 25)
+//		{
+//			DebugPrint("#ICT: Sending!");
+//			IComms_Transmit(&txMsg);
+//			testTxCounter =0;
+//		}
+
+		throttleTxCounter++;
+		if (throttleTxCounter == THROTTLE_RATE) {
+			DebugPrint("#ICT: Sending Throttle!");
+
+			iCommsMessage_t throttleTxMsg;
+			throttleTxMsg.standardMessageID = 0x0001;
+			throttleTxMsg.dataLength = 2;
+			percentage_t p = SystemGetThrottlePercentage();
+			throttleTxMsg.data[0] = p;
+			throttleTxMsg.data[1] = p >> 8 & 0xFF;
+
+			DebugPrint("data[0] = %x", throttleTxMsg.data[0]);
```

Comment:
Use the tags, we're developing a serial monitor that will organize output streams based on the tags. this file uses #ICT: (it should be defined as a constant, if not you can added it)